rust/noosphere-gateway/src/jobs/worker_queue/worker.rs

```diff
@@ -0,0 +1,151 @@
+use super::Processor;
+use anyhow::{anyhow, Result};
+use std::{fmt::Debug, time::SystemTime};
+use tokio::{
+    sync::mpsc::{unbounded_channel, UnboundedSender},
+    task::JoinHandle,
+    time::Duration,
+};
+
+pub(crate) type WorkerResponse<J> = (usize, Result<Option<J>>);
+
+/// A job with additional metadata to handle time outs and retries.
+pub struct JobRequest<P: Processor> {
+    pub job: P::Job,
+    pub attempt: usize,
+    pub start_time: Option<SystemTime>,
+}
+
+impl<P> JobRequest<P>
+where
+    P: Processor,
+{
+    /// Creates a new [JobRequest].
+    pub fn new(job: P::Job) -> Self {
+        Self {
+            job,
+            attempt: 0,
+            start_time: None,
+        }
+    }
+
+    /// Marks [JobRequest] as a failed attempt, clearing
+    /// its last start time, and returns whether this job
+    /// should be retried or not.
+    pub fn mark_attempt_failed(&mut self, retries: usize) -> bool {
+        self.start_time = None;
+        self.attempt < retries
+    }
+}
+
+impl<P> Debug for JobRequest<P>
+where
+    P: Processor,
+{
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("JobRequest")
+            .field("job", &self.job)
+            .field("attempts", &self.attempt)
+            .finish()
+    }
+}
+
+/// Provides an interface to perform a job via [Processor]
+/// on another thread.
```

Comment:
Can you comment about the usage and lifecycle of a `Worker` in the documentation?